plugins/pulp_rpm/plugins/importers/yum/purge.py

```diff
@@ -263,3 +264,28 @@ def get_remote_units(file_function, tag, process_func):
     finally:
         file_handle.close()
     return remote_named_tuples
+
+
+def remove_unit_duplicate_nevra(unit_key, type_id, repo_id):
```

Comment:
this would work *only* with units that have NEVRA, and those are only rpm/srpm and drpm. Other yum types have completely different unit key.